src/libraries/Common/src/Interop/Unix/System.Native/Interop.MemAlloc.cs

```diff
@@ -8,8 +8,17 @@ internal static partial class Interop
 {
     internal static partial class Sys
     {
+        [DllImport(Interop.Libraries.SystemNative, EntryPoint = "SystemNative_MemAlignedAlloc")]
+        internal static extern IntPtr MemAlignedAlloc(nuint alignment, nuint size);
```

Comment:
I used `IntPtr` here for consistency, but these should really be `void*` IMO.

Likewise, I think it would make sense to call the P/Invoke `aligned_alloc`, `malloc`, etc, since they are 1-to-1 with the C export.

Reply:
I'll do it in a follow up since doing so would require cleanup of `Marshal` as well